app/services/api/v1/message_service.rb

```diff
@@ -0,0 +1,18 @@
+module Api
+  module V1
+    class MessageService
+      def initialize(message_params, current_user, conversation_id)
+        @message_params = message_params
+        @current_user = current_user
+        @conversation_id = conversation_id
+      end
+
+      def create
+        conversation = Conversation.find(@conversation_id)
+        raise InvalidConversationError unless conversation.user1_id == @current_user.id || conversation.user2_id == @current_user.id
```

Comment:
maybe it's not invalid, should more like a unauthorized thing.. wdyt?
Also you could check for invalid if the Conversation did not exists